third_party/blink/renderer/core/brave_session_cache.cc

```diff
@@ -124,6 +125,28 @@ bool AllowFontFamily(ExecutionContext* context,
   return true;
 }
 
+int FarbledInteger(ExecutionContext* context,
+                   FarbleKey key,
+                   int spoof_value,
+                   int min_value,
+                   int max_value) {
+  BraveSessionCache& cache = BraveSessionCache::From(*context);
+  return cache.FarbledInteger(key, spoof_value, min_value, max_value);
+}
+
+bool AllowScreenFingerprinting(ExecutionContext* context) {
+// Don't apply this protection on mobile.
+#if BUILDFLAG(IS_ANDROID) || BUILDFLAG(IS_IOS)
+  return true;
+#else
+  return !base::FeatureList::IsEnabled(
+             blink::features::kBraveBlockScreenFingerprinting) ||
+         !context ||
```

Comment:
shouldn't this be the same conditions as `AllowFingerprinting` plus also checking the feature?

Reply:
also in general not or conditions are hard to parse imo. I think it would be better to split them up with separate return statements

Reply:
Good point; I cleaned it up.